lib/mumuki/classroom/models/concerns/course_member.rb

```diff
@@ -0,0 +1,54 @@
+module CourseMember
+  extend ActiveSupport::Concern
+
+  included do
+    include Mongoid::Timestamps
+
+    field :uid, type: String
+    field :first_name, type: String
+    field :last_name, type: String
+    field :image_url, type: String
+    field :name, type: String
+    field :email, type: String
+    field :social_id, type: String
+    field :organization, type: String
+    field :course, type: Mumukit::Auth::Slug
+
+    create_index({organization: 1, course: 1, uid: 1}, {unique: true})
+  end
+
+  def as_user(verified: true)
+    member_json = as_json.merge_if(verified, validated_first_name: first_name, validated_last_name: last_name)
+    User.whitelist_attributes member_json
+  end
+
+  class_methods do
+    def ensure_not_exists!(query)
+      existing_members = where(query)
+      return unless existing_members.exists?
+      raise Mumuki::Classroom::CourseMemberExistsError, {existing_members: existing_members.map(&:uid)}.to_json
```

Comment:
The return-raise flow of this method is a bit far-fetched. Why not just `raise ... unless ...`